covariancematrix.go

```diff
@@ -80,3 +80,43 @@ func CovarianceMatrix(cov *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.De
 	cov.Scale(1/(n-1), cov)
 	return cov
 }
+
+// CorrelationMatrix calculates a correlation matrix from a matrix of data,
+// using a two-pass algorithm. The matrix returned will be symmetric, square,
+// and positive-semidefinite.
+//
+// The weights wts should have the length equal to the number of rows in
+// input data matrix x. cov should either be a square matrix with the same
+// number of columns as the input data matrix x, or nil in which case a new
+// Dense matrix will be constructed.  Weights cannot be negative.
+func CorrelationMatrix(corr *mat64.Dense, x mat64.Matrix, wts []float64) *mat64.Dense {
+
+	// TODO(jonlawlor): indicate that the resulting matrix is symmetric, and change
+	// the returned type from a *mat.Dense to a *mat.Symmetric.
+
+	// This will panic if the sizes don't match, or if wts is the wrong size.
+	corr = CovarianceMatrix(corr, x, wts)
+	Cov2Corr(corr)
+	return corr
+}
+
+// Cov2Corr converts a covariance matrix to a correlation matrix.
```

Comment:
I made it exported because a user might want to convert between the two, and may not have the sample data used to generate the covariance.  Both R and Matlab define it (in R core, and in the Financial Toolbox in Matlab), so I figure there is a need.  I can implement the inverse Corr2Cov if you want.